### PR TITLE
Update to Multicall V3

### LINKED ISSUE #511: Update Multicall contracts to Multicall V3
https://github.com/mds1/multicall

### PR DESCRIPTION
**Pull request by Uxio0**
- Update Multicall addresses
- Try deterministic address for Multicall if network is not supported
- Closes #511

### CODE REVIEW

File: gnosis/eth/multicall.py [hunk 1]
```diff
@@ -36,23 +37,35 @@ class MulticallDecodedResult:
...
+        EthereumNetwork.FANTOM_TESTNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.FANTOM_OPERA: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.BINANCE_SMART_CHAIN_MAINNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.BINANCE_SMART_CHAIN_TESTNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.KOVAN: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.RINKEBY: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.KCC_MAINNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.KCC_TESTNET: "0x665683D9bd41C09cF38c3956c926D9924F1ADa97",
+        EthereumNetwork.ROPSTEN: "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696",
+        EthereumNetwork.CELO: "0xcA11bde05977b3631167028862bE2a173976CA11",
```
Comment (moisses89) [inline gnosis/eth/multicall.py:66]: There are 3 diferent CELO networks :smile: but no one is named only CELO

**Review by moisses89 — CHANGES_REQUESTED** [review]

File: gnosis/eth/multicall.py [hunk 2]
```diff
@@ -36,23 +37,35 @@ class MulticallDecodedResult:
...
+        EthereumNetwork.AVALANCHE_FUJI_TESTNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.FANTOM_TESTNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.FANTOM_OPERA: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.BINANCE_SMART_CHAIN_MAINNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.BINANCE_SMART_CHAIN_TESTNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.KOVAN: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.RINKEBY: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.KCC_MAINNET: "0xcA11bde05977b3631167028862bE2a173976CA11",
+        EthereumNetwork.KCC_TESTNET: "0x665683D9bd41C09cF38c3956c926D9924F1ADa97",
+        EthereumNetwork.ROPSTEN: "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696",
```
Comment (moisses89) [inline gnosis/eth/multicall.py:65]: Where did you find this addreses?

File: gnosis/eth/multicall.py [hunk 3]
```diff
@@ -65,14 +78,17 @@ def __init__(
         ethereum_network = ethereum_client.get_network()
         address = multicall_contract_address or self.ADDRESSES.get(ethereum_network)
         if not address:
-            raise EthereumNetworkNotSupported(
-                "Multicall contract not available for %s", ethereum_network.name
-            )
+            # Try with Multicall V3 deterministic address
+            address = self.ADDRESSES.get(EthereumNetwork.MAINNET)
```
Comment (moisses89) [inline gnosis/eth/multicall.py:82]: always mainnet?

Comment (Uxio0) [reply gnosis/eth/multicall.py:82]: It's a deterministic address and they tried to use MAINNET address everywhere

File: gnosis/eth/multicall.py (hunk 1 above)
Comment (Uxio0) [reply gnosis/eth/multicall.py]: Fixed!

File: gnosis/eth/multicall.py (hunk 2 above)
Comment (Uxio0) [reply gnosis/eth/multicall.py:65]: https://github.com/mds1/multicall

Comment (moisses89) [reply gnosis/eth/multicall.py:65]: Ok, following the list that you shared Ropsten is not the correct address and I think that also KCC TESTNET had the incorrect address configured.

Comment (Uxio0) [reply gnosis/eth/multicall.py:65]: Right on Ropsten, KCC has the previous address as Multicall V3 was not deployed there

Comment (Uxio0) [reply gnosis/eth/multicall.py]: Fixed

**Review by moisses89 — APPROVED** [review]

**Pull request merged**